src/libsystemd/sd-bus/sd-bus.c

```diff
@@ -2897,7 +2897,7 @@ static int bus_process_internal(sd_bus *bus, bool hint_priority, int64_t priorit
 
         /* We don't allow recursively invoking sd_bus_process(). */
         assert_return(!bus->current_message, -EBUSY);
-        assert(!bus->current_slot);
+        assert_return(!bus->current_slot, -EBUSY);
```

Comment:
this is unnecessary... the slot can only be non-NULL if the message is non-NULL too. Thus, the first assert_return() protects us against recursive invocations by the caller, but the second assert() just catches memory corruptions/our own code bugs, as the slot must be NULL anyway if the message is NULL, which we just verified.

Reply:
Right. I reverted the change and instead added a comment based on what you wrote.